src/main/kotlin/primitives/Tree.kt

```diff
@@ -0,0 +1,35 @@
+package primitives
+
+import iterator.TreeIterator
+
+abstract class Tree<K : Comparable<K>, V, N : Node<K, V, N>> : Iterable<V> {
+    var root: N? = null
```

Comment:
пользователь библиотеки не должен иметь прямого доступа к корню дерева
```suggestion
    protected var root: N? = null
```

Reply:
Поменял на protected и сделал в тестах доступ к этому полю через рефлексию